### PR TITLE
Feature/dashboard header

### LINKED ISSUE #255: Dashboard - Header
image

image

- [ ] Gradient background
- [ ] Coding coach logo in top right corner
- [ ] "Modal" type layover for new messages
- [ ] "Open" button which leads to the messages panel
- [ ] "Let's go" button which leads to the mentorships panel

## Text

**Text for account with messages**
"You have X new messages!"

**Text for account with messages (Button)**
"Open"

**Text for account with no messages (Button)**
"Let's go"

**Text for account with no messages (Mentor)**
"Welcome back! Head over to the mentorships tab to find a new mentor or mentee!"

**Text for account with no messages (Mentee)**
"Welcome back! Head over to the mentorships tab to find a new mentor!"

# Assets
All icons & color codes can be found on the Sketch cloud: https://sketch.cloud/s/xVR5A/p/dashboard

### PR DESCRIPTION
**Pull request by tureey**
# Details
This PR refers to the Dashboard's Header Component

## Description
Create a Header component

## Issue
https://github.com/Coding-Coach/coding-coach/issues/255

Depends on #259 
Closes #255

### CODE REVIEW

**Comment by tureey** [comment]
**I am working in progress.**

Things left:
* Wait for the components PR #259 to be merged so I can use the media included in
* Generate the translation for `Dashboard` (I executed compile and extract and still not working) @crysfel  Any idea?
* Use the same `spacing size` for the padding of the Header and the `width` not used by the notification
* Component tests

**Do you have any idea for the third point?**
_I want to add some `padding: X` to an element in both sides and to another Component (notification/card) give a `width: calc(100% - 2*(this_padding_size))` so both elements can be aligned_

**Comment by emmawedekind** [comment]
I have messaged @steoneill regarding the status of #259. If I receive no response within a day or so I'll re-assign. Can you continue once that is merged @tureey?

File: src/pages/dashboard/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import logoSrc from './assets/logo.svg';
+
+function TopHeader() {
+  return (
+    <div className={classNames('bg-gradient-primary h-48 py-5', 'md:py-6 md:h-64 lg:py-8')}>
+      <div className="flex justify-between items-center w-5/6 m-auto">
+        <span className={classNames('text-white text-2xl font-bold')}>
+          <Trans id="dashboard.header.title" />
+        </span>
+        <img src={logoSrc} className="w-12" />
```
Comment (crysfel) [inline src/pages/dashboard/components/Header.js]: Please use the `Image` component that we have

File: src/pages/dashboard/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <React.Fragment>
+      <header>
+        {/* This includes the Section name and the Logo of CodingCoach */}
+        <TopHeader />
+      </header>
+
+      {banner && (
+        <div
+          style={{ marginTop: '-6rem' }}
+          className={classNames('mx-auto w-5/6 p-8', 'bg-white shadow', 'md:w-3/5 xl:w-2/5')}
```
Comment (crysfel) [inline src/pages/dashboard/components/Header.js]: ahhhh... maybe these where in different lines? If so use just use backticks instead.

File: src/pages/dashboard/components/IllustratedMessage.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function Banner({ imageSrc, message, button }) {
+  return (
+    <Fragment>
+      <img className="w-6" src={imageSrc} />
+
+      <span className={classNames('block text-secondary-dark text-lg font-light my-4')}>
```
Comment (crysfel) [inline src/pages/dashboard/components/IllustratedMessage.js]: please remove `classNames` is not needed here

File: src/pages/dashboard/components/IllustratedMessage.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function Banner({ imageSrc, message, button }) {
+  return (
+    <Fragment>
+      <img className="w-6" src={imageSrc} />
```
Comment (crysfel) [inline src/pages/dashboard/components/IllustratedMessage.js]: Use the `Image` component we have

File: src/pages/dashboard/Dashboard.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import Header from './components/Header';
+import IllustratedMessage from './components/IllustratedMessage';
+import Button from 'components/button/Button';
+import LogoSrc from './components/assets/logo.svg';
+
+class Dashboard extends Component {
+  getBanner = () => (
+    <IllustratedMessage
+      imageSrc={LogoSrc}
+      message="You have 3 new mentorship messages!"
```
Comment (crysfel) [inline src/pages/dashboard/Dashboard.js]: i18n?

File: src/pages/dashboard/Dashboard.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      button={
+        <Button onClick={() => console.log('click')} size="small">
+          open
```
Comment (crysfel) [inline src/pages/dashboard/Dashboard.js]: i18n?

File: src/pages/dashboard/Dashboard.stories.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import Dashboard from './Dashboard';
+
+storiesOf('Dashboard', module).add('Header', () => {
```
[the remark this answers is not in the document]
Comment (crysfel) [reply src/pages/dashboard/Dashboard.stories.js]: I don't think we need storybook for the pages... I feel like storybook should only be for components, thoughts?

File: src/containers/App/App.js [hunk 8]
```diff
@@ -2,14 +2,15 @@ import React from 'react';
...
-import { Home, Login } from 'pages';
+import { Home, Login, Dashboard } from 'pages';
 
 const App = () => {
   return (
     <I18nProvider language="en" catalogs={catalogs}>
       <Router>
         <Home path="/" />
         <Login path="/auth/login" />
+        <Dashboard path="/dashboard" />
```
[the remark this answers is not in the document]
Comment (crysfel) [reply src/containers/App/App.js]: No, reach-router support something like this:

```
<PrivateViews path="/app">
  <Dashboard path="dashboard" />
</PrivateViews>
```

Inside `PrivateViews` we are going to render the actual content or the login page, depending if the user is logged in or not. But that's not part of this task, for now just render the content inside `PrivateViews`.

File: src/pages/dashboard/Dashboard.js (hunk 5 above)
Comment (tureey) [reply src/pages/dashboard/Dashboard.js]: Well this should not even be here. This is an example hardcoded. I guess this "banner" or "Dashboard notifications" or "dashboard messages" will be get through redux. 

I just wanted to point out how to use it.. maybe its better to do this on the story and then remove from the view for now? @crysfel

**Comment by tureey** [comment]
Why github still keep saying of conflicts? I pulled `develop`, merged into my `branch`.
I had to resolve some conflicts and now I'm updated

**Comment by crysfel** [comment]
That's weird... did you `rebase` your branch?

**Comment by tureey** [comment]
oh.. I did not! 😆  @crysfel

**Comment by emmawedekind** [comment]
Any update @tureey ? :)

**Comment by tureey** [comment]
**I think all the feedback given has been corrected. Do another check please! @crysfel **

The **only pending thing is to update the logo I'm currently using** _(the white version for the dark)_.
Still pending because https://github.com/Coding-Coach/coding-coach/pull/291 is not closed yet.

I even prepared to make the "banner/notification or whatever" optional. So if we don't pass it the header is rendered without it.

The point is to pass this Banner or the name we decide with a given props provided by redux. (This component is done, we will just need to think about this banner combinations of icon/text)

_see pictures attached and give feedback please!_

**Comment by emmawedekind** [comment]
Looks so great @tureey! My only critique is that we need the gradient on the background to not be so white on the right side (the logo fades into it). Do you need a gradient from me?

**Comment by emmawedekind** [comment]
Also when the user is on a wider view port, we should make the header text larger ;)

File: src/pages/dashboard/components/IllustratedMessage.js (hunk 3 above)
Comment (crysfel) [reply src/pages/dashboard/components/IllustratedMessage.js]: you would use `classNames` when you want to dynamically set styles

**Comment by emmawedekind** [comment]
> @emmawedekind I'll look for which gradient I used. But yes, you can provide me one if needed!
> 
> Related tot he text, you mean the _Dashboard_ text? or the text in the card? I'll update it tomorrow

The Dashboard text :)

**Comment by crysfel** [comment]
Build is failing 😬

**Comment by tureey** [comment]
Yes, I haven't invest time since I am waiting for a gradient from @emmawedekind and trent to merge the icons PR @crysfel

**Comment by tureey** [comment]
I've updated it.

- Updated the logo. Now we use the dark one
- Changed the `IllustratedMessage` API
- Incremented sizes of text and logo

If desired by @emmawedekind still need to update the gradient.

Should we wait for it? Should we merge and open a new issue to update the gradient once she provides us it? @crysfel 

PS: The build in local is working OK for me...

**Comment by crysfel** [comment]
Just fix the issues in travis and we should merge!

There are some linting issues:

```
./src/pages/auth/PrivateViews.js
  Line 1:  'React' is defined but never used  no-unused-vars
./src/pages/dashboard/components/Header.js
  Line 5:  'logoSrc' is defined but never used  no-unused-vars
  Line 6:  'Image' is defined but never used    no-unused-vars```

Just remove those unused variables and it should be good to go.

Thanks

**Comment by tureey** [comment]
Tomorrow at first in the morning I'll do it. (it means in 12h) @crysfel

**Comment by tureey** [comment]
@crysfel it's strange because the `imgSrc` and `<Image>` are used..
my build in local works. And my precommit as well.

Can you confirm it's still happening for you @crysfel ?

**Comment by crysfel** [comment]
Mmmm... I'm not really sure, we are running the linter locally as well 🤔

**Pull request merged**